http-server-netty/src/main/java/io/micronaut/http/server/netty/MicronautHttpData.java

```diff
@@ -513,7 +513,7 @@ public String getValue() throws IOException {
 
         @Override
         public void setValue(String value) throws IOException {
-            throw new UnsupportedOperationException();
+            setContent(Unpooled.copiedBuffer(value.toCharArray(), factory.characterEncoding));
```

Comment:
no need for the toCharArray

Reply:
Ahh, great, thanks. I somehow missed the `CharSequence` variation.

[c5b0e8d](https://github.com/micronaut-projects/micronaut-core/pull/10601/commits/c5b0e8dcc5a481c316c592e6fed66232dcfa8ffb)